### PR TITLE
Implementation for faster-whisper

### PR DESCRIPTION
**Pull request by ArrangingFear56**
Updates whisper(extras) to use faster-whisper instead of openAI-whisper, defaults to large-v3-turbo model, install CUDA 12 and cuDNN 8, then run `pip install -r whisperRequirements.txt` to finish installation.

### CODE REVIEW

File: modules/speech_recognition/whisper_module.py [hunk 1]
```diff
@@ -1,57 +1,56 @@
-"""
```
Comment (Cohee1207) [inline modules/speech_recognition/whisper_module.py]: Adding files via upload messes up the diff due to line endings mismatch. Please resubmit using proper authoring tools. Use GitHub Codespaces if you need an online environment.

File: modules/speech_recognition/whisper_module.py [hunk 2]
```diff
@@ -13,24 +13,24 @@
         - oobabooga text-generation-webui github: https://github.com/oobabooga/text-generation-webui
 """
 from flask import jsonify, abort, request
-
-import whisper
+from faster_whisper import WhisperModel
```
Comment (Cohee1207) [inline modules/speech_recognition/whisper_module.py]: I would like it to give a choice between the regular whisper and faster_whisper. Requiring new dependencies for already existing users is not ideal. Either provide a command line option, or at least wrap the import in try/except.

File: modules/speech_recognition/whisper_module.py [hunk 3]
```diff
@@ -13,24 +13,24 @@
...
 
 DEBUG_PREFIX = "<stt whisper module>"
 RECORDING_FILE_PATH = "stt_test.wav"
 
-model = None
+model_size = "large-v3-turbo"
+
+model = WhisperModel(model_size, device="cuda", compute_type="float16")
```
Comment (Cohee1207) [inline modules/speech_recognition/whisper_module.py]: This binds the model to cuda. Consider the cases when cuda is not available or not desired (using ARM CPU / AMD GPU / wanting to converve VRAM).

File: modules/speech_recognition/whisper_module.py [hunk 4]
```diff
@@ -45,9 +45,10 @@ def process_audio():
         file = request.files.get('AudioFile')
         language = request.form.get('language', default=None)
         file.save(RECORDING_FILE_PATH)
-
-        result = model.transcribe(RECORDING_FILE_PATH, condition_on_previous_text=False, language=language)
-        transcript = result["text"]
+        segments, info = model.transcribe(RECORDING_FILE_PATH, beam_size=5)
```
Comment (Cohee1207) [inline modules/speech_recognition/whisper_module.py]: Does it preserve the ability to set a language for the transcription? Automatic detection may lack accuracy.

**Review by Cohee1207 — CHANGES_REQUESTED** [review]

File: modules/speech_recognition/whisper_module.py (hunk 2 above)
Comment (ArrangingFear56) [reply modules/speech_recognition/whisper_module.py]: added argument `--use-faster-whisper` to enable faster-whisper and defaults to the old module.

File: modules/speech_recognition/whisper_module.py (hunk 3 above)
Comment (ArrangingFear56) [reply modules/speech_recognition/whisper_module.py]: added argument `--faster-whisper-cpu` to switch to cpu

File: modules/speech_recognition/whisper_module.py (hunk 4 above)
Comment (ArrangingFear56) [reply modules/speech_recognition/whisper_module.py]: It should, at least according to faster-whisper's original repo, difficult to test though.

File: README.md [hunk 5]
```diff
@@ -230,6 +230,19 @@ cd SillyTavern-extras
 | `--sd-remote-ssl`        | Use SSL for the remote SD backend<br>Default: **False** |
 | `--sd-remote-auth`       | Specify the `username:password` for the remote SD backend (if required) |
 
+## For faster-whisper instead of whisper when using whisper(extras)
+1. Install CUDA 12 and cuDNN 8
+2. Install faster-whisper requirements
+```
+pip install -r fasterWhisperRequirements.txt
```
Comment (Cohee1207) [inline README.md]: Please rename the file to match the rest of the requirements files (e.g. `requirements-faster-whisper.txt`)

Comment (ArrangingFear56) [reply README.md:237]: done

**Review by Cohee1207 — APPROVED** [review]
Tested. Works fine, thanks.

**Pull request merged**